### ISSUE TITLE
Federation API spec does not declare fields as requiredfor /_matrix/key/v2/server

### ISSUE THREAD

**Issue by valkum** [matrix-org/matrix-doc#2508]
in https://https://matrix.org/docs/spec/server_server/r0.1.3#get-matrix-key-v2-server-keyid does not declare the response fields as required but synapse fails with a KeyError when at least on of the fields is not present.

**Comment by turt2live** [comment]
as mentioned in the conduit room: we probably just forgot to add the "required" label here.

**Comment by valkum** [comment]
Yeah just put that here as a reminder that it gets fixed sometime.
